base/abstractarray.jl

```diff
@@ -1490,7 +1490,7 @@ AbstractVecOrTuple{T} = Union{AbstractVector{<:T}, Tuple{Vararg{T}}}
 
 _typed_vcat_similar(V, ::Type{T}, n) where T = similar(V[1], T, n)
 _typed_vcat(::Type{T}, V::AbstractVecOrTuple{AbstractVector}) where T =
```

Comment:
Can we make this change just for `V::Tuple` and keep using `mapreduce` for `V::AbstractVector`, since in that case, it is probably beneficial to avoid the extra allocation?

Reply:
Since `vcat` itself is allocating I don't think this will make any noticeable difference, but it's easy to add if there's a good argument for why it's still needed.